src/userbase-server/db.js

```diff
@@ -584,8 +606,12 @@ const putTransaction = async function (transaction, userId, databaseId) {
     else throw new Error(`Failed to put transaction with ${e}.`)
   }
 
+  // username is put on the transaction for transmitting,
+  // but not for storing.
+  transaction['username'] = (await userPromise).username
```

Comment:
Does this make sense? Or should I be sending over the username by some other means?